maintenance/wikia/WikiFactoryVariables/setGlobalBasedOnWam.class.php

```diff
@@ -0,0 +1,113 @@
+		$this->addOption( 'varValue', 'What value should I set the variable to?', true, true );
+		$this->addOption( 'dependency', 'Should I consider any other variable\'s value?', false, true );
+		$this->addOption( 'dependencyValue', 'What should be a value of the dependant variable?', false, true );
+		$this->addOption( 'dry', 'Would you like to try it out first?', false, false );
+	}
+
+	public function execute() {
+		/**
+		 * Get options
+		 */
+		$limit = (int)$this->getOption( 'limit' );
+		$var = $this->getOption( 'var' );
+		$varValue = $this->getOption( 'varValue' );
+		$dependency = $this->getOption( 'dependency' );
+		$dependencyValue = $this->getOption( 'dependencyValue', true );
+		$dry = $this->getOption( 'dry' );
+
+		/**
+		 * If a dependency is set - this array will be empty and you will get $limit of wikias
+		 * only based on the WAM rank.
+		 */
+		$wikisToOrder = [];
+
+		if ( $dependency !== null ) {
+			/**
+			 * Convert the value to boolean or integer.
+			 * Leave as a string otherwise.
+			 */
+			$dependencyData = WikiFactory::getVarByName( $dependency );
+
+			if ( $dependencyData->cv_variable_type === 'boolean' ) {
+				$dependencyValue = filter_var( $dependencyValue, FILTER_VALIDATE_BOOLEAN );
+			} elseif ( $dependencyData->cv_variable_type === 'integer' ) {
+				$dependencyValue = (int)$dependencyValue;
+			}
+			$dependencyId = $dependencyData->cv_id;
+
+			$wikisToOrder = WikiFactory::getCityIDsFromVarValue( $dependencyId, $dependencyValue, '=' );
+
+			$count = count( $wikisToOrder );
+			if ( $count === 0 ) {
+				$this->output( "No wikias found with {$dependency}={$dependencyValue}.\n\n" );
+				return;
+			}
+
+			$this->output( "\nFound {$count} wikias with {$dependency}={$dependencyValue}.\n\n" );
+		}
+
+		$dataMartService = new DataMartService();
+		$wikiIdsToConvert = $dataMartService->getWikisOrderByWam( $limit, $wikisToOrder );
+		$count = count( $wikiIdsToConvert );
+		$this->output( "Got {$count} wikis to convert ordered by WAM rank.\n\n" );
+
+
+		/**
+		 * Convert the value to boolean or integer.
+		 * Leave as a string otherwise.
+		 */
+		$varData = WikiFactory::getVarByName( $var );
+		if ( $varData->cv_variable_type === 'boolean' ) {
+			$varValue = filter_var( $varValue, FILTER_VALIDATE_BOOLEAN );
+		} elseif ( $varData->cv_variable_type === 'integer' ) {
+			$varValue = (int)$varValue;
+		}
+		$varId = (int)$varData->cv_id;
+
+		foreach ( $wikiIdsToConvert as $wikiId ) {
+			$this->output( "Setting {$var} to {$varValue} for wikiId={$wikiId}\n\n" );
+
+			if ( !$dry ) {
+				WikiFactory::setVarById( $varId, $wikiId, $varValue );
+			}
+		}
+
+		if ( !$dry ) {
+			$this->output( "All done! Let's validate the result...\n\n" );
+
+			$success = $fail = [ ];
+
+			foreach ( $wikiIdsToConvert as $wikiId ) {
+				if ( WikiFactory::getVarValueByName( $var, $wikiId ) === $varValue ) {
+					$success[] = $wikiId;
+				} else {
+					$fail[] = $wikiId;
```

Comment:
I want to verify if all values are actually returned properly, not just the transaction status... I figured out that it could pass in a script that you only run from time to time. :)